### PR TITLE
DistanceWeightedGraph Utility method 

### LINKED ISSUE #159: Add utilities to represent a grid as graph
For several application cases (validation, automated layouting ...) having a grid represented by their vertex-edge-graph would be very helpful. As we are already making use of the JGraphT library this one could be used with own vertex-edge representations. 

For a first go I would propose the following two graph structures:

- DistanceWeightedGraph where the distance is the edge weight

As JGraphT allows only one weight per edge type, both graphs needs to become their distinct implementation. 

If you have any other suggestions on useful graph structures feel free to add them in the comments.

### PR DESCRIPTION
**Pull request by johanneshiry**
Resolves #159 

Adds a utility method to create a DistanceWeightedGraph from a GridContainer

### CODE REVIEW

File: src/main/java/edu/ie3/datamodel/utils/ContainerUtils.java [hunk 1]
```diff
@@ -16,18 +15,125 @@
...
+    } catch (NullPointerException | IllegalArgumentException | UnsupportedOperationException ex) {
+      log.error("Error adding 3 winding transformer edges to graph: ", ex);
+      return Optional.empty();
+    }
+
+    // if we reached this point, we can safely return a valid graph
+    return Optional.of(graph);
+  }
+
+  private static void addDistanceGraphEdge(
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/utils/ContainerUtils.java:125]: Huhu! JavaDoc, where are you?

File: src/test/groovy/edu/ie3/datamodel/utils/ContainerUtilsTest.groovy [hunk 2]
```diff
@@ -293,6 +294,23 @@ class ContainerUtilsTest extends Specification {
...
+		when:
+		Optional<DistanceWeightedGraph> resultingGraphOpt = ContainerUtils.getDistanceTopologyGraph(grid)
+
+		then:
+		resultingGraphOpt.isPresent()
+		DistanceWeightedGraph resultingGraph = resultingGraphOpt.get()
+
+		resultingGraph.vertexSet() == ComplexTopology.grid.getRawGrid().getNodes()
+
+		resultingGraph.edgeSet().size() == 7
```
Comment (ckittl) [inline src/test/groovy/edu/ie3/datamodel/utils/ContainerUtilsTest.groovy:310]: I would also expect a test for `ContainerUtils#addDistanceGraphEdge(DistanceWeightedGraph, NodeInput, NodeInput)`, as this is _the_ crucial point in this concept.

File: src/main/java/edu/ie3/datamodel/utils/ContainerUtils.java [hunk 3]
```diff
@@ -16,18 +15,125 @@
...
 /** Offers functionality useful for grouping different models together */
 public class ContainerUtils {
+
+  private static final Logger log = LoggerFactory.getLogger(ContainerUtils.class);
+
   private ContainerUtils() {
     throw new IllegalStateException("Utility classes cannot be instantiated");
   }
 
+  /**
```
[the remark this answers is not in the document]
Comment (johanneshiry) [reply src/main/java/edu/ie3/datamodel/utils/ContainerUtils.java:36]: WTF?! 👀

**Pull request merged**